`__{a}exit__` typing

### Things to check first

- [X] I have checked that my issue does not already have a solution in the [FAQ](https://apscheduler.readthedocs.io/en/master/faq.html)

- [X] I have searched the existing issues and didn't find my bug already reported there

- [X] I have checked that my bug is still present in the latest release


### Version

4.0.0a4

### What happened?

Passing an `AsyncScheduler` instance to `AsyncExitStack.enter_async_context` results in:

```
src/infra/__init__.py:61: error: Argument 1 to "enter_async_context" of "AsyncExitStack" has incompatible type "AsyncScheduler"; expected "AbstractAsyncContextManager[AsyncScheduler]"  [arg-type]
src/infra/__init__.py:61: note: Following member(s) of "AsyncScheduler" have conflicts:
src/infra/__init__.py:61: note:     Expected:
src/infra/__init__.py:61: note:         def __aexit__(self, type[BaseException] | None, BaseException | None, TracebackType | None, /) -> Coroutine[Any, Any, bool | None]
src/infra/__init__.py:61: note:     Got:
src/infra/__init__.py:61: note:         def __aexit__(self, type[BaseException], BaseException, TracebackType, /) -> Coroutine[Any, Any, None]
```

To conform to the ABC the args received by the exit method must be union with `None`.

### How can we reproduce the bug?

```py
    async with AsyncExitStack() as stack:
        scheduler = AsyncScheduler()
        scheduler = await stack.enter_async_context(scheduler)
```

--- a/src/apscheduler/_schedulers/async_.py
+++ b/src/apscheduler/_schedulers/async_.py
@@ -169,9 +169,9 @@ class AsyncScheduler:
 
     async def __aexit__(
         self,
-        exc_type: type[BaseException],
-        exc_val: BaseException,
-        exc_tb: TracebackType,
+        exc_type: type[BaseException] | None,
+        exc_val: BaseException | None,
+        exc_tb: TracebackType | None,
     ) -> None:
         await self.stop()
         await self._exit_stack.__aexit__(exc_type, exc_val, exc_tb)

--- a/src/apscheduler/_schedulers/sync.py
+++ b/src/apscheduler/_schedulers/sync.py
@@ -124,9 +124,9 @@ class Scheduler:
 
     def __exit__(
         self,
-        exc_type: type[BaseException],
-        exc_val: BaseException,
-        exc_tb: TracebackType,
+        exc_type: type[BaseException] | None,
+        exc_val: BaseException | None,
+        exc_tb: TracebackType | None,
     ) -> None:
         if self._exit_stack:
             self._exit_stack.__exit__(exc_type, exc_val, exc_tb)

